pandas/core/arrays/_arrow_string_mixins.py

```diff
@@ -49,7 +53,19 @@ def _str_pad(
         elif side == "right":
             pa_pad = pc.utf8_rpad
         elif side == "both":
-            pa_pad = pc.utf8_center
+            if pa_version_under17p0:
+                # GH#59624 fall back to object dtype
+                from pandas import array
+
+                obj_arr = self.astype(object, copy=False)  # type: ignore[attr-defined]
+                obj = array(obj_arr, dtype=object)
+                result = obj._str_pad(width, side, fillchar)  # type: ignore[attr-defined]
+                return type(self)._from_sequence(result, dtype=self.dtype)  # type: ignore[attr-defined]
```

Comment:
Not sure how significant this is, but one small performance disadvantage of this instead of `type(self)(pa.array(result))` is that `_from_sequence` has an (in this case) unnecessary call to `lib.ensure_string_array`.

The problem is that we don't know here which pyarrow type to use? But then could do something like `pa.array(result, type=self.dtype.pyarrow_dtype)` to preserve the type?

Reply:
im not inclined to worry about this too much since its in a branch that will be going away eventually